src/analysis/retail/monk/mistweaver/modules/spells/Upwelling.tsx

```diff
@@ -203,8 +217,12 @@ class Upwelling extends Analyzer {
         this.masteryOverhealing += event.overheal || 0;
         this.masteryAbsorbed += event.absorbed || 0;
       }
-      this.masteryTickTock = !this.masteryTickTock;
+    } else {
+      if (!this.masteryTickTock) {
```

Comment:
This does not need to be accounted for. You were never accounting for base gusts procs in any original EF calculation so you do not need to subtract it from additional mastery you would get from extra duration hots